packages/firestore/src/platform_browser/webchannel_connection.ts

```diff
@@ -215,6 +212,19 @@ export class WebChannelConnection implements Connection {
       sendRawJson: true,
       supportsCrossDomainXhr: true
     };
+
+    // Send our custom headers as a '$httpHeaders=' url param to avoid CORS
+    // preflight round-trip. This is formally defined here:
+    // https://github.com/google/closure-library/blob/b0e1815b13fb92a46d7c9b3c30de5d6a396a3245/closure/goog/net/rpc/httpcors.js#L40
+    //
+    // For some unclear reason (see
```

Comment:
At a high level I understand what this comment is saying but I don't understand mechanically how this is actually working.

When we specify `httpHeadersOverwriteParam: '$httpHeaders'`, which headers are getting tucked into that variable? If we omit setting that, where do they get set instead?